### DISCUSSION TITLE
Fan without a preset

### DISCUSSION THREAD

**Discussion by ant-car** [Koenkk/zigbee2mqtt#18613] [Devices]
Hi all,

@lucasteligioridis and I are looking to add support for the [clipsal fan controller ](https://www.clipsal.com/products/detail?CatNo=41ECSFWMZ&itemno=41ECSFWMZ-VW) but it doesn't have an on/auto or smart mode.  it's values are off, low , medium and high which causes the [preset array to be empty](https://github.com/Koenkk/zigbee2mqtt/blob/master/lib/extension/homeassistant.ts#L499) and [fail this assertion](https://github.com/Koenkk/zigbee2mqtt/blob/master/lib/extension/homeassistant.ts#L524). 

Any ideas on how to get around this issue?  We could make one of the values 'on' but there's no logical value that would relate to on.

**Comment by lucasteligioridis** [comment]
@ant-car this issue is not a problem now by remapping `on` to another value / speed in a converter, see how this can be achieved here:
```
    fan_mode: {
        ...tz.fan_mode,
        convertSet: async (entity, key, value, meta) => {
            // Remap on to medium because device does not support it
            // https://github.com/Koenkk/zigbee2mqtt/issues/18620
            if (value.toLowerCase() === 'on') value = 'medium';
            return tz.fan_mode.convertSet(entity, key, value, meta);
        },
    }
```

Full context and gist of the entire converter in question is in the issue: 
https://github.com/Koenkk/zigbee2mqtt/issues/18620
